build/rules.mak

```diff
@@ -58,6 +59,18 @@ $(MODULE_DIR)/%: %.mako $(BUILD_HELPER_SCRIPTS) $(METADATA_FILES)
 	$(call trace_to_console, "Generating",$@)
 	$(_hide_cmds)$(call log_command,$(call GENERATE_SCRIPT, $<, $(dir $@), $(METADATA_DIR)))
 
+$(MODULE_DIR)/%_pb2.py: %.proto
+	$(call trace_to_console, "Generating",$@ and $(notdir $*)_pb2_grpc.py)
+	$(_hide_cmds)$(call log_command,mkdir -p $(OUTPUT_DIR)/build/proto/$(dir $*))
+	$(_hide_cmds)$(call log_command,python -m grpc_tools.protoc $(addprefix -I=,$(METADATA_DIRS)) --python_out=$(MODULE_DIR) --grpc_python_out=$(MODULE_DIR) $*.proto)
+	$(_hide_cmds)$(call log_command,sed -i 's/^import session_pb2/from . import session_pb2/' $(MODULE_DIR)/$*_pb2*.py)
```

Comment:
FYI, `protoc` can generate `from a.b import c` automatically, but only if the package names are set up to match Python module names.

Reply:
Yeah, I was mostly thinking about the compatibility side of it. Or maybe I shouldn't care about that, since grpc-device is breaking all sorts of compatibility anyway?

Reply:
I meant to reply here rather than open a separate issue. I think I deleted the other one now. We should desire namespace compatibility, as each breaking change is considered separately. Most of this is internal to the module though right?

Reply:
Sorry, I misremembered the behavior. It uses relative filesystem paths, not the package name. If you organize your `.proto` files in the same hierarchy as your Python module and change the `import` directives to use paths that are relative to the root of the hierarchy, I think you can get it to work without `sed`. This is a source compatibility break but not a wire compatibility break.

For example, if I change `generated\nidaqmx\nidaqmx.proto` to do `import "source/protobuf/session.proto";` and I run `python -m grpc_tools.protoc --proto_path=. generated\nidaqmx\nidaqmx.proto --python_out=.`, then it generates `generated\nidaqmx\nidaqmx_pb2.py` containing `from source.protobuf import session_pb2 as source_dot_protobuf_dot_session__pb2`.

Related `protoc` bugs (with links to more): https://github.com/protocolbuffers/protobuf/issues/10329 https://github.com/protocolbuffers/protobuf/issues/5374

Anyway, this is just an FYI 😀